File: 27. TurnOffBitsLSBToI.cpp
// Topic: Turn off bits from LSB to ith bit inclusive
#include <bits/stdc++.h>
using namespace std;

int turnOffLSBtoI(int n, int i)
{
    int mask = ~((1 << (i + 1)) - 1);
    return n & mask;
}

int main()
{
    cout << "Turn off LSB..1 of 13: " << turnOffLSBtoI(13, 1) << '\n';
    return 0;
}

// Time Complexity: O(1)
// Space Complexity: O(1)

File: 27. TurnOffBitsLSBToI.py
# Topic: Turn off bits from LSB to ith bit inclusive

def turn_off_lsb_to_i(n, i):
    mask = ~((1 << (i+1)) - 1)
    return n & mask

if __name__ == '__main__':
    print('Turn off LSB..1 of 13:', turn_off_lsb_to_i(13,1))

# Time Complexity: O(1)
# Space Complexity: O(1)
